scipy/stats/_fit.py

```diff
@@ -720,7 +720,7 @@ def nlpsf(free_params, data=data):  # bind data NOW
 
 
 def goodness_of_fit(dist, data, *, known_params=None, fit_params=None,
-                    guessed_params=None, statistic='ad', n_mc_samples=9999,
+                    fit_init_params=None, statistic='ad', n_mc_samples=9999,
```

Comment:
I think the rest is almost ready. I'll have a few comments when I have time, but I think we can merge that pretty quickly. The changed parameter name will be a sticking point, though.

Reply:
I really don't insist on it. I just would have made my first reading easier. The real purpose on this particular parameter is revealed in the last sentence of its description.

Reply:
I renamed it back to `guessed_params` in https://github.com/scipy/scipy/pull/17506/commits/7dfa0efad8efbb3c65e5e9cb5aef93c495ffb050.

For me, `known_parameters` and `fit_params` have another quality than `guessed_params`. The first two really matter for the goodness of fit test. In an ideal world with perfect optimizers, the latter would not be needed at all. It has nothing to do with the GoF test per se, but is merely a numerical, computational detail to help the optimizer.

Reply:
Hmm, there may be a misunderstanding. That distinction is important, but it's really only `known_parameters` that affects the results. In a world with perfect optimizers (and a perfect fitting objective function - MLE has issues) we would expose neither `fit_params` nor `guessed_params`. The only reason I exposed `fit_params` is because it is reasonable to expect that the user will have already fitted the distribution to data before running this test, but it is unlikely to change the results meaningfully if `goodness_of_fit` performs the fit for the `fit_params`. 

That said, one could make an argument for removing `fit_params` entirely to protect users from themselves. If the user does a very good job of fitting `fit_params`, it makes the test more conservative (higher p-values). If the user does a poor job of fitting `fit_params` (e.g. if they hand-tune parameters), it makes the test less conservative (lower p-values). One could argue that the exact same numerical procedure should be used to fit `data` as the rest of the Monte Carlo samples. Perhaps let's merge this and I'll add a note about that separately.